bsi/src/main/java/org/roaringbitmap/bsi/longlong/Roaring64NavigableMapSliceIndex.java

```diff
@@ -0,0 +1,710 @@
+    }
+  }
+
+  private void grow(int newBitDepth) {
+    int oldBitDepth = this.bA.length;
+
+    if (oldBitDepth >= newBitDepth) {
+      return;
+    }
+
+    Roaring64NavigableMap[] newBA = new Roaring64NavigableMap[newBitDepth];
+    if (oldBitDepth != 0) {
+      System.arraycopy(this.bA, 0, newBA, 0, oldBitDepth);
+    }
+
+    for (int i = newBitDepth - 1; i >= oldBitDepth; i--) {
+      newBA[i] = new Roaring64NavigableMap();
+      newBA[i].setSerializationMode(serializationMode);
+      if (this.runOptimized) {
+        newBA[i].runOptimize();
+      }
+    }
+    this.bA = newBA;
+  }
+
+  public void setValues(List<Pair<Long, Long>> values) {
+    if (values == null || values.isEmpty()) return;
+    long maxValue =
+        values.stream().mapToLong(Pair::getRight).filter(Objects::nonNull).max().getAsLong();
+    long minValue =
+        values.stream().mapToLong(Pair::getRight).filter(Objects::nonNull).min().getAsLong();
+    ensureCapacityInternal(minValue, maxValue);
+    for (Pair<Long, Long> pair : values) {
+      setValueInternal(pair.getKey(), pair.getValue());
+    }
+  }
+
+  /**
+   * merge will merge 2 bsi into current
+   * merge API was designed for distributed computing
+   * note: current and other bsi has no intersection
+   *
+   * @param otherBsi other bsi we need merge
+   */
+  public void merge(Roaring64NavigableMapSliceIndex otherBsi) {
+
+    if (null == otherBsi || otherBsi.ebM.isEmpty()) {
+      return;
+    }
+
+    // todo whether we need this
+    if (Roaring64NavigableMap.intersects(this.ebM, otherBsi.ebM)) {
+      throw new IllegalArgumentException("merge can be used only in bsiA  bsiB  is null");
+    }
+
+    int bitDepth = Integer.max(this.bitCount(), otherBsi.bitCount());
+    Roaring64NavigableMap[] newBA = new Roaring64NavigableMap[bitDepth];
+    for (int i = 0; i < bitDepth; i++) {
+      Roaring64NavigableMap current = i < this.bA.length ? this.bA[i] : new Roaring64NavigableMap();
+      Roaring64NavigableMap other =
+          i < otherBsi.bA.length ? otherBsi.bA[i] : new Roaring64NavigableMap();
+      newBA[i] = Roaring64NavigableMap.or(current, other);
+      newBA[i].setSerializationMode(serializationMode);
+      if (this.runOptimized || otherBsi.runOptimized) {
+        newBA[i].runOptimize();
+      }
+    }
+    this.bA = newBA;
+    this.ebM.or(otherBsi.ebM);
+    this.ebM.setSerializationMode(serializationMode);
+    this.runOptimized = this.runOptimized || otherBsi.runOptimized;
+    this.maxValue = Long.max(this.maxValue, otherBsi.maxValue);
+    this.minValue = Long.min(this.minValue, otherBsi.minValue);
+  }
+
+  @Override
+  public Roaring64NavigableMapSliceIndex clone() {
+    Roaring64NavigableMapSliceIndex bitSliceIndex = new Roaring64NavigableMapSliceIndex();
+    bitSliceIndex.minValue = this.minValue;
+    bitSliceIndex.maxValue = this.maxValue;
+    bitSliceIndex.ebM = this.ebM.clone();
+    bitSliceIndex.ebM.setSerializationMode(serializationMode);
```

Comment:
Why do you need to call `setSerializationMode ` after `clone`?

Reply:
redundant code, will remove this